.github/workflows/documentation.yml

```diff
@@ -0,0 +1,33 @@
+name: Build documentation
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - 'docs/**'
+      - 'mkdocs.yml'
+  pull_request:
+    paths:
+      - 'docs/**'
+      - 'mkdocs.yml'
+
+jobs:
+  build:
+    name: Build documentation
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: mkdocs build --strict
+      run: |
+        docker run --rm -v ${PWD}:/docs squidfunk/mkdocs-material build --strict --config-file docs/mkdocs.yml
+  deploy:
+    name: Deploy documentation to gh-pages
+    runs-on: ubuntu-20.04
+    if: github.event_name != 'pull_request'
```

Comment:
Why is this `!=`? I saw this usage when referencing conditional jobs on my own workflow I created and it seemed odd. I used `github.event_name == 'push'` to be more explicit that the job is only for `push` event, no other event should trigger it.

Reply:
Thinking of it again it makes actually no difference as the workflow is only triggered by two events: 
1. push
2. pull_request
So it is enough to define a condition on the deploy stage with `!= pull_request` as the only option that is left is `push`.

But as you splitted the workflows now it doesn't matter anymore ✅